docsk8s/quick-start/ls-k8s-configuration-files.asciidoc

```diff
@@ -0,0 +1,284 @@
+[[ls-k8s-configuration-files]]
+=== Logstash configuration files in Kubernetes
+
+WARNING: This documentation is still in development. This feature may be changed or removed in a future release.
+
+The <<ls-k8s-configuration-files,guide to {ls} configuration and settings files in Kubernetes>> guide demonstrates how to configure logstash, including pipeline configuration, in Kubernetes.
+
+* <<qs-pipeline-configuration>>
+* <<qs-logstash-yaml>>
+* <<qs-jvm-options>>
+* <<qs-logging>>
+
+Configuring Logstash includes two types of configuration files: _pipeline configuration files_, which define the Logstash processing pipeline, and _settings files_ which specify options that control Logstsah startup and execution.
+{logstash-ref}/config-setting-files.html[Config Settings Files] contains information on these files, and this guide explains how these map to a Kubernetes configuration.
+
+[float]
+[[qs-pipeline-configuration]]
+=== Pipeline Configuration
+
+This section will explain how to configure single and multiple pipeline logstash configurations. Note that this section does not describe how to set this up using {logstash-ref}/logstash-centralized-pipeline-management.html[Centralized Pipeline Management]
+
+
+[float]
+[[qs-single-pipeline-config]]
+==== Single Pipeline
+
+The existing docker image contains a default `pipeline.yml`, which expects a single pipeline, with the definition of that pipeline present in `/usr/share/logstash/pipeline`, as either a single file or collection of files, typically defined as a `ConfigMap` or series of `ConfigMaps` - note that
+a single Kubernetes `ConfigMap` has a size limit of 1MB.
+
+
+This example contains a simple pipeline definition, with the inputs and outputs split into separate configuration files:
+
+
++
+[source,yaml]
+--
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: logstash-pipeline <1>
+  labels:
+    app: logstash-demo
+data:
+  logstash-input.conf: | <2>
+    input {
+      beats {
+        port => "5044"
+      }
+    }
+  logstash-output.conf: | <3>
+    output {
+      elasticsearch {
+        hosts => ["https://demo-es-http:9200"]
+      }
+    }
+--
+
+<1> Name of `ConfigMap` to be referenced in `Deployment`.
+<2> This will create a `ConfigMap` representing the inputs for a pipeline.
+<3> This will create a `CongigMap` representing the outputs for a pipeline.
+
+From there, you will define your `Volume` in your `Deployment` template
+
+[source,yaml]
+--
+volumes:
+  - name: logstash-pipeline
+    configMap:
+      name: logstash-pipeline
+--
+
+and mount the volume in your container
+
+[source,yaml]
+--
+volumeMounts:
+    - name: logstash-pipeline
+      mountPath: /usr/share/logstash/pipeline
+--
+
+
+[float]
+[[qs-multiple-pipeline-config]]
+==== Multiple Pipelines
+
+Multiple pipelines <insert link> require a `ConfigMap` to represent pipelines.yml, where pipeline configurations can be created inline, or in separate `configMap` files or folders.
```

Comment:
```suggestion
{ls} uses the `pipelines.yml` file to define {logstash-ref}/multiple-pipelines.html[multiple pipelines]. 
{ls} in {k8s} requires a `ConfigMap` to represent the content that would otherwise be in `pipelines.yml`.
You can create pipeline configurations inline, or in separate `configMap` files or folders. 
```